src/gilded_rose.js

```diff
@@ -13,50 +13,82 @@ items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 80));
 items.push(new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20));
 items.push(new Item('Conjured Mana Cake', 3, 6));
 
+var newItems = []
+
+let itemInventory = {'Aged Brie': increasesOverTime,
+'Backstage passes to a TAFKAL80ETC concert': increasesMoreTowardsDeadlineAndExpires,
+'Sulfuras, Hand of Ragnaros': () => {},
+'Conjured Mana Cake': doubleDecreases,
+'+5 Dexterity Vest': regularDecrease}
+
 function update_quality() {
-  for (var i = 0; i < items.length; i++) {
-    if (items[i].name != 'Aged Brie' && items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-      if (items[i].quality > 0) {
-        if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-          items[i].quality = items[i].quality - 1
-        }
-      }
-    } else {
-      if (items[i].quality < 50) {
-        items[i].quality = items[i].quality + 1
-        if (items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].sell_in < 11) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-          if (items[i].sell_in < 6) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-        }
-      }
-    }
-    if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-      items[i].sell_in = items[i].sell_in - 1;
-    }
-    if (items[i].sell_in < 0) {
-      if (items[i].name != 'Aged Brie') {
-        if (items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].quality > 0) {
-            if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-              items[i].quality = items[i].quality - 1
-            }
-          }
-        } else {
-          items[i].quality = items[i].quality - items[i].quality
-        }
-      } else {
-        if (items[i].quality < 50) {
-          items[i].quality = items[i].quality + 1
-        }
-      }
-    }
+  items.forEach((item) => {updateItemQuality(item)})
+}
+
+function updateItemQuality(item) {
+  if(typeof itemInventory[item.name] !== 'function') {
```

Comment:
Use of typeOf and InstanceOf should be avoided. Use of these operators is often a sign that your code is violating the [Liskov substitution principle ](https://en.wikipedia.org/wiki/Liskov_substitution_principle)